javalink/ref.py

```diff
@@ -259,20 +276,30 @@ def initialize_package_list(app):
             app.verbose('[javalink] %s', traceback.format_exc())
 
 
-def parse_docroot(srcdir, root):
+def parse_docroot(srcdir, default_version, root):
```

Comment:
This should just return the full dictionary now, instead of a tuple. In that case, `normalize_docroot` is a better name.

Also, I think the signature `normalize_docroot(app, root)` is more appropriate now since you need 2 values from app: `app.env.srcdir` and `app.config.javalink_default_version`